core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -190,6 +197,27 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
                     }
 
                     @Override
+                    public void onAllNodesAcked(@Nullable Throwable t) {
+                        // the cluster state that includes the newly created index has been acknowledged,
+                        // now wait for the write consistency number of shards to be allocated before returning,
+                        // as that is when indexing operations can take place on the newly created index
+                        final long timeToTimeout = request.ackTimeout().nanos() - (System.nanoTime() - startTimeNS);
+                        if (timeToTimeout <= 0L) {
+                            // we've already timed out
+                            listener.onResponse(newResponse(false));
+                        } else {
+                            final TimeValue remainingTime = TimeValue.timeValueNanos(timeToTimeout);
+                            try {
+                                IndexCreationMonitor monitor = new IndexCreationMonitor(settings, clusterService, threadContext);
```

Comment:
ok, I see now that we create one monitor per index creation. Is that necessary? We could just have a single IndexCreationMonitor instance as the class is pretty stateless.